File: sol/962/962.py
class Solution:
    def maxWidthRamp(self, nums: List[int]) -> int:
        n = len(nums)
        st = []
        for i, x in enumerate(nums):
            if len(st) == 0 or nums[st[-1]] > x:
                st.append(i)
        res = 0
        for j in range(n - 1, -1, -1):
            while len(st) != 0 and nums[st[-1]] <= nums[j]:
                res = max(res, j - st.pop())
        return res

File: sol/962/962.cpp
struct Solution {
    int maxWidthRamp(vector<int>& nums) {
        int n = nums.size(), res = 0;
        vector<int> st;
        for (int i = 0; i < n; ++i){
            if (st.empty() || nums[st.back()] > nums[i]){
                st.push_back(i);
            }
        }
        for (int j = n - 1; j >= 0; --j){
            while (!st.empty() && nums[st.back()] <= nums[j]){
                res = max(res, j - st.back());
                st.pop_back();
            }
        }
        return res;
    }
};
